libarchive/archive_windows.c

```diff
@@ -641,13 +641,14 @@ __la_waitpid(HANDLE child, int *status, int option)
 	(void)option;/* UNUSED */
 	do {
 		if (GetExitCodeProcess(child, &cs) == 0) {
-			CloseHandle(child);
 			la_dosmaperr(GetLastError());
+			CloseHandle(child);
 			*status = 0;
 			return (-1);
 		}
 	} while (cs == STILL_ACTIVE);
 
+	CloseHandle(child);
```

Comment:
(Personally,) I'm surprised that `waitpid` would guarantee that the child process handle was closed. I can't find prior art for the actual non-Windows `waitpid` to do so, but process handles aren't tangible on those platforms.